fail2ban/config.sls

```diff
@@ -1,55 +1,104 @@
+# -*- coding: utf-8 -*-
+# vim: ft=sls
+
 {% from "fail2ban/map.jinja" import fail2ban with context %}
 
-include:
-  - fail2ban.deprecated
-  - fail2ban
-
-{{ fail2ban.prefix }}/etc/fail2ban/fail2ban.local:
-{% if fail2ban.config %}
-  file.managed:
-    - source: salt://fail2ban/files/fail2ban_conf.template
-    - template: jinja
-    - context:
-        config:
-            Definition: {{ fail2ban.config|yaml }}
+fail2ban.config.fail2ban:
+{% if fail2ban.config is defined %}
+
+{% set fail2ban_config = fail2ban.config.get('source_path',
+                           'salt://fail2ban/files/config.jinja') %}
```

Comment:
can't this path be stored in one of the jinja map? *(if it has any sense)*

Reply:
Definitely, there are a number of things that can be improved in the promoted `ng` files.  I've purposefully kept this PR a straight conversion, without mixing in a refactor at the same time.  All of these improvements can be tackled in due course.